File: Java/GBAddressBuilder.java
public class GBAddressBuilder extends AddressBuilder {

    public GBAddressBuilder(AddressData addressData) {
        super(addressData);
        titleF = "Ms";
        titleM = "Mr";
    }

    @Override
    public void buildLine3(){
        address.addLine(addressData.city() + "\n");
    }

    @Override
    public void buildLine4(){
        address.addLine(addressData.zip() + "\n");
    }

    @Override
    public void buildLine5(){
        address.addLine(addressData.state());
    }
}

File: TypeScript/GBAddressBuilder.ts
import { AddressBuilder } from './AddressBuilder';
import { AddressData } from './AddressData';

export class GBAddressBuilder extends AddressBuilder{

  buildLine3() : void {
    this.address.addLine(this.addressData.city + "\n");
  }

  buildLine4() : void {
    this.address.addLine(this.addressData.zip + "\n");
  }

  buildLine5() : void {
    this.address.addLine(this.addressData.state);
  }

  constructor(addressData: AddressData) {
    super(addressData);
    this.title_f = "Ms";
    this.title_m = "Mr";
  }
}